pylint/extensions/typing.py

```diff
@@ -219,6 +224,66 @@ def visit_attribute(self, node: nodes.Attribute) -> None:
         if self._should_check_callable and node.attrname == "Callable":
             self._check_broken_callable(node)
 
+    @only_required_for_messages("redundant-union-assign-typehint")
+    def visit_annassign(self, node: nodes.AnnAssign) -> None:
+        type_annotations = list(node.annotation.get_children())
```

Comment:
I don't like the use of `get_children()`. It's easy to miss a case when accessing children by index and much more difficult to understand compared to normal attribute access.